### PR TITLE
Increase Max Partition Count from 64 to 128

### PR DESCRIPTION
**Pull request by nibanks**
The bulk of this change comes down to handling the +64 processor scenario. In this case, we need to set the proper group affinity before setting the processor index. Determining the mapping to these values is quite involved. The platform now generates and persists a global table for this information to be used efficiently later.

Only Windows user mode is updated here.

### CODE REVIEW

File: src/platform/platform_winuser.c [hunk 1]
```diff
@@ -21,6 +21,8 @@
 uint64_t QuicPlatformPerfFreq;
 uint64_t QuicTotalMemory;
 QUIC_PLATFORM QuicPlatform = { NULL };
+QUIC_PROCESSOR_INFO* QuicProcessorInfo;
```
Comment (ThadHouse) [inline src/platform/platform_winuser.c:24]: Don't we need to free this in QuicPlatformUninitialize?

Comment (nibanks) [reply src/platform/platform_winuser.c:24]: Already done in the latest commit. :)

File: src/platform/platform_winuser.c [hunk 2]
```diff
@@ -55,6 +57,142 @@ QuicPlatformSystemUnload(
...
+                        goto FindNumaNode;
+                    }
+                    ProcessorOffset += Info->Group.GroupInfo[i].ActiveProcessorCount;
+                }
+            }
+            Offset += Info->Size;
+        }
+
+        //printf("Failed to determine group!\n");
+        goto Error;
```
Comment (ThadHouse) [inline src/platform/platform_winuser.c:156]: If we hit this, we leak QuicNumaMasks

File: src/core/quicdef.h [hunk 3]
```diff
@@ -105,7 +105,7 @@ typedef struct QUIC_PATH QUIC_PATH;
 //
 // Maximum number of partitions to support.
 //
-#define QUIC_MAX_PARTITION_COUNT                64
+#define QUIC_MAX_PARTITION_COUNT                128
```
Comment (ThadHouse) [inline src/core/quicdef.h:108]: Any specific reasons now to limit this to 128? EPYC is now at 256 threads in a dual socket system, and over the next few years a setup like that will just get cheaper and more common.

File: src/platform/datapath_winuser.c [hunk 4]
```diff
@@ -577,15 +577,14 @@ QuicDataPathInitialize(
...
     DatapathLength =
         sizeof(QUIC_DATAPATH) +
-        Sysinfo.dwNumberOfProcessors * sizeof(QUIC_DATAPATH_PROC_CONTEXT);
+        MaxProcCount * sizeof(QUIC_DATAPATH_PROC_CONTEXT);
 
     //
     // Thread affinity uses a bit mask. We can't support more cores than bits.
     //
-    QUIC_FRE_ASSERT(Sysinfo.dwNumberOfProcessors <= sizeof(DWORD_PTR) * 8);
+    QUIC_FRE_ASSERT(MaxProcCount <= sizeof(DWORD_PTR) * 8);
```
Comment (ThadHouse) [inline src/platform/datapath_winuser.c:587]: Wouldn't this assertion still fail on more then 64 cores?

Comment (nibanks) [reply src/platform/datapath_winuser.c:587]: Fixed.

File: src/core/quicdef.h (hunk 3 above)
Comment (nibanks) [reply src/core/quicdef.h:108]: We currently use a `uint8_t` for partition count, so `256` doesn't fix. We can change it to a `uint16_t` in a future change.

**Pull request merged**